ui/apps/platform/src/utils/searchUtils.ts

```diff
@@ -253,3 +253,37 @@ export function convertToExactMatch(item): unknown {
     }
     return `r/^${item}$`;
 }
+
+/**
+ * Adds acs regex flag to values in the searchFilter object
+ *
+ * All values are prefixed by default
+ * If keysToTransform is provided, only those keys will be modified
+ *
+ * @param {Object} searchFilter Original searchFilter object
+ * @param {Array<string>} [keysToTransform] Optional – The keys in the searchFilter object to transform
+ * @returns {Object} New SearchFilter object where values (determined by keysToTransform) are prefixed with 'r/'
+ */
+export function addRegexPrefixToFilters(
+    searchFilter: SearchFilter,
+    keysToTransform: string[] | null = null
+) {
+    const modifiedFilter: SearchFilter = {};
+
+    Object.keys(searchFilter).forEach((key) => {
+        const value = searchFilter[key];
+        const shouldTransform = !keysToTransform || keysToTransform.includes(key);
```

Comment:
Two questions, not necessarily change requests:
* Is there a use case for default `null` versus not calling the function if not needed? That is, could `keysToTransform` argument be required. It can still have `[]` value if built at runtime.
* If there is a use case, what are pro and con for `[]` instead of `null` as default value?
* If there is a use case, what are pro and con of short circuit return of unchanged `searchFilter` if no keys?

Reply:
Maybe it's a little confusing but not passing in `keysToTransform` will actually transform all keys. Based off how it's currently written if you defaulted `keysToTransform` to an empty array or passed in an empty array, it wouldn't transform any keys.

I just ran this test:
``` javascript
myFilter = {
    field_one: 'test',
    field_two: 'another',
    field_three: ['one', 'two', 'three']
}

addRegexPrefixToFilters(myFilter)
// {
//   field_one: 'r/test',
//   field_two: 'r/another',
//   field_three: [ 'r/one', 'r/two', 'r/three' ]
// }

addRegexPrefixToFilters(myFilter, ['field_one'])
// {
//   field_one: 'r/test',
//   field_two: 'another',
//   field_three: [ 'one', 'two', 'three' ]
// }

addRegexPrefixToFilters(myFilter, ['field_one', 'field_three'])
// {
//   field_one: 'r/test',
//   field_two: 'another',
//   field_three: [ 'r/one', 'r/two', 'r/three' ]
// }

addRegexPrefixToFilters(myFilter, [])
// {
//   field_one: 'test',
//   field_two: 'another',
//   field_three: [ 'one', 'two', 'three' ]
// }
```